cirq/contrib/routing/greedy.py

```diff
@@ -0,0 +1,253 @@
+# Copyright 2019 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import itertools
+import random
+from typing import (Any, Callable, cast, Dict, Hashable, Iterable, List,
+                    Mapping, Optional, Set, Tuple)
+
+import numpy as np
+import networkx as nx
+
+from cirq import circuits, ops
+import cirq.contrib.acquaintance as cca
+from cirq.contrib.routing.initialization import get_initial_mapping
+from cirq.contrib.routing.swap_network import SwapNetwork
+from cirq.contrib.routing.utils import (get_timeslices,
+                                        get_ops_consistency_with_device_graph)
+
+SWAP = cca.SwapPermutationGate()
+
+
+def route_circuit_greedily(circuit: circuits.Circuit, device_graph: nx.Graph,
+                           **kwargs) -> SwapNetwork:
+    """Greedily routes a circuit on a given device.
+
+    Args:
+        circuit: The circuit to route.
+        device_graph: The device's graph, in which each vertex is a qubit and
+            each edge indicates the ability to do an operation on those qubits.
+
+    See GreedyRouter for argument details.
+    """
+
+    router = GreedyRouter(circuit, device_graph, **kwargs)
+    router.route()
+
+    swap_network = router.swap_network
+    swap_network.circuit = circuits.Circuit.from_ops(
+        swap_network.circuit.all_operations(), device=swap_network.device)
+    return swap_network
+
+
+class GreedyRouter:
```

Comment:
Add brief docstrings to all of these functions or show example usage in the class docstring